snippets/oadp-image-stream-tag-trigger.adoc

```diff
@@ -0,0 +1,16 @@
+
+//This snippet appears in the following assemblies:
+//
+//backup_and_restore/application_backup_and_restore/backing_up_and_restoring/restoring-applications.adoc
+//OADP 1.4.1 Release Notes
+
+:_mod-docs-content-type: SNIPPET
+
+During a File System Backup (FSB) restore operation, a `Deployment` resource referencing an `ImageStream` is not restored properly. Due to this, pod restoration `postHook` is not executed, and the volume content is not restored. 
+
+This happens because, during the restore operation, an OpenShift controller updates the `spec.template.spec.containers[0].image` field in the `Deployment` resource with the `ImageStreamTag` hash. The update triggers the rollout of a new pod, affecting the FSB volume restoration and pod restoration `postHook` execution. For more information, see "Triggering updates on image stream changes".
+
+The workaround for this behavior is a two-step restore process:
+
+* Perform a restore by excluding the `deploymentconfig` resources, for example `--exclude-resources=deployment.apps`.
+* Perform a restore by including these resources, for example, `--include-resources=deployment.apps`.
```

Comment:
````suggestion
During a File System Backup (FSB) restore operation, a `Deployment` resource referencing an `ImageStream` is not restored properly. Restored Pod running File System Backup as well as `postHook` is killed prematurely due to a rollout triggerd by ImageStreamTag update.

This happens because, during the restore operation, OpenShift controller updates the `spec.template.spec.containers[0].image` field in the `Deployment` resource with an updated `ImageStreamTag` hash. The update triggers the rollout of a new pod, killing the pod velero is running File System Backup and post restore hook on. For more information about image stream trigger, see "Triggering updates on image stream changes".

The workaround for this behavior is a two-step restore process:

* First perform a restore excluding the `Deployment` resources, for example
```
velero restore create <RESTORE_NAME> \
  --from-backup <BACKUP_NAME> \
  --exclude-resources=deployment.apps
```
* Once first restore is successful, perform a second restore by including these resources, for example
```
velero restore create <RESTORE_NAME> \
  --from-backup <BACKUP_NAME> \
  --include-resources=deployment.apps
```
````

Reply:
Thanks for the review. I reworded a little to use 'terminating' instead of 'killing' :). Thanks.